packages/contracts/contracts/universal/MultiSigWallet.sol

```diff
@@ -0,0 +1,601 @@
+        require(isOwner[_addr]);
+        _;
+    }
+
+    /**
+     * @notice Ensure that the transaction exists.
+     *
+     * @param _transactionId Index of submitted transaction requested.
+     */
+    modifier transactionExists(uint256 _transactionId) {
+        require(transactions[_transactionId].destination != address(0));
+        _;
+    }
+
+    /**
+     * @notice Ensure that the transaction with id and owner is confirmed.
+     *
+     * @param _transactionId Index of submitted transaction requested.
+     * @param _owner         Address resource requested.
+     */
+    modifier confirmed(uint256 _transactionId, address _owner) {
+        require(confirmations[_transactionId][_owner]);
+        _;
+    }
+
+    /**
+     * @notice Ensure that the transaction with id and owner is not confirmed.
+     *
+     * @param _transactionId Index of submitted transaction requested.
+     * @param _owner         Address resource requested.
+     */
+    modifier notConfirmed(uint256 _transactionId, address _owner) {
+        require(!confirmations[_transactionId][_owner]);
+        _;
+    }
+
+    /**
+     * @notice Ensure that the transaction not executed.
+     *
+     * @param _transactionId Index of submitted transaction requested.
+     */
+    modifier notExecuted(uint256 _transactionId) {
+        require(!transactions[_transactionId].executed);
+        _;
+    }
+
+    /**
+     * @notice Ensure that the address is not zero address.
+     *
+     * @param _address Address resource requested.
+     */
+    modifier notNull(address _address) {
+        require(_address != address(0));
+        _;
+    }
+
+    /**
+     * @notice Ensure that the number of confirmations required is valid.
+     *
+     * @param _ownerCount               A number of owners.
+     * @param _numConfirmationsRequired Number of required confirmations.
+     */
+    modifier validNumConfirmations(uint256 _ownerCount, uint256 _numConfirmationsRequired) {
+        require(
+            _numConfirmationsRequired <= _ownerCount
+            && _numConfirmationsRequired != 0
+            && _ownerCount != 0);
+        _;
+    }
+
+    /**
+     * @notice Initializer.
+     *
+     * @param _owners                   List of initial owners.
+     * @param _numConfirmationsRequired Number of required confirmations.
+     */
+    function initialize(address[] memory _owners, uint256 _numConfirmationsRequired) public onlyInitializing {
+        require(_owners.length > 0, "MultiSigWallet: owners requred");
+        for (uint256 i = 0; i < _owners.length;) {
+            address owner = _owners[i];
+            require(!isOwner[owner], "MultiSigWallet: owner already exists");
+            require(owner != address(0), "MultiSigWallet: invalid owner address");
+            isOwner[owner] = true;
+
+            unchecked {
+                ++i;
+            }
+        }
+        owners = _owners;
+        numConfirmationsRequired = _numConfirmationsRequired;
```

Comment:
Don't we need to check `_numConfirmationsRequired` is not 0?

Reply:
You're right.